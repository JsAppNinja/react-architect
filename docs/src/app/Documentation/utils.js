export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
