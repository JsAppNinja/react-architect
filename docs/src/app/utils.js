import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
