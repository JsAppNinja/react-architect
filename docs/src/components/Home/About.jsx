import React from 'react';
import { bem } from 'react-md';

const About = () => (
  <h3 className={bem('home', 'about', {}, 'md-text-container')}>
    This project&apos;s goal is to be able to create a fully accessible material design
    styled website using React Components and Sass. With the separation of styles in
    Sass instead of inline styles, it should hopefully be easy to create custom
    components with the existing styles.
  </h3>
);

export default About;
