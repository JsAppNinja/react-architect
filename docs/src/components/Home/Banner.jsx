import React from 'react';
import { Link } from 'react-router-dom';
import Button from 'react-md/lib/Buttons/Button';

import InlineSVG from 'components/InlineSVG';
import logo from '!!raw-loader!./logo.svg';

const Banner = () => (
  <header className="home__banner">
    <h1 className="md-display-2">react-md</h1>
    <InlineSVG src={logo} className="home__logo" />
    <Button
      raised
      secondary
      type={null}
      to="/components"
      component={Link}
    >
      View Demo
    </Button>
  </header>
);

export default Banner;
