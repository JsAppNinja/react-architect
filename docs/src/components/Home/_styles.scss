@import '~globals';

@include style-logo-to-theme($md-primary-color, $md-secondary-color, true);

.home {
  &__banner {
    align-items: center;
    background: $md-primary-color;
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 400px;
    padding: 2em;

    .md-display-2 {
      color: $md-white-base;
    }
  }

  &__about {
    line-height: $md-line-height;
    padding: 1em $md-default-padding 3em;
    white-space: initial;
  }

  &__svg-card {
    text-decoration: none;
  }

  &__logo {
    margin-bottom: 2em;

    tspan {
      font-family: 'Montserrat';
      font-weight: $md-font-bold;
    }
  }

  &__footer {
    margin-left: auto;
    margin-right: auto;
    max-width: 1200px;
  }
}
