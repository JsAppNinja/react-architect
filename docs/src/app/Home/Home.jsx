import React, { Component, PropTypes } from 'react';
import PureRenderMixin from 'react-addons-pure-render-mixin';
import { RaisedButton } from 'react-md';

import './_home.scss';
import * as components from '../components';
import { toDashedName } from '../Documentation/utils';

const firstLink = 'components/' + toDashedName(Object.keys(components)[0]);

export default class Home extends Component {
  constructor(props) {
    super(props);

    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
  }

  static propTypes = {
    history: PropTypes.object, // from react-router
  };

  componentDidMount() {
    this.appBar = document.querySelector('.react-md-docs-toolbar');
    this.appBar.classList.add('no-shadow');
    window.addEventListener('scroll', this.updateToolbar);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.updateToolbar);
    this.appBar.classList.remove('no-shadow');
  }

  updateToolbar = () => {
    const isEnabled = window.scrollY < 600;
    const isClassed = this.appBar.classList.contains('no-shadow');
    if(isEnabled && !isClassed) {
      this.appBar.classList.add('no-shadow');
    } else if(!isEnabled && isClassed) {
      this.appBar.classList.remove('no-shadow');
    }
  };

  viewDemo = () => {
    this.props.history.pushState(null, `/${firstLink}`);
  };

  render() {
    return (
      <div className="home">
        <section className="banner">
          <h1 className="md-display-2">react-md</h1>
          <h4 className="md-title">Material Design inspired React components built with sass</h4>
          <RaisedButton onClick={this.viewDemo} secondary label="Demo" />
        </section>
        <section className="about">
        </section>
        <section className="getting-started">
        </section>
      </div>
    );
  }
}
