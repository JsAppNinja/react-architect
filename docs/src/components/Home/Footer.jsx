import React from 'react';
import PropTypes from 'prop-types';
import { GridList, bem } from 'react-md';

import gettingStarted from './getting-started.svg';
import customization from './customization.svg';
import components from './components.svg';
import SVGCard from './SVGCard';

const base = 'home';

const Footer = ({ style }) => (
  <GridList component="footer" style={style} className={bem(base, 'footer')} cellClassName={bem(base, 'svg-card')}>
    <SVGCard
      to="getting-started"
      src={gettingStarted}
      alt="A Mac and a Macbook with design mock ups."
      title="Getting Started"
    />
    <SVGCard
      to="customization"
      src={customization}
      alt="A chemistry set with material design colors flowing into a text editor."
      title="Customization"
    />
    <SVGCard
      to="components"
      src={components}
      alt="A chemistry set with material design colors flowing into a text editor."
      title="Components"
    />
  </GridList>
);

Footer.propTypes = {
  style: PropTypes.object,
};

export default Footer;
