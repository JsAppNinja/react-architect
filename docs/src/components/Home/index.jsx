import React from 'react';

import Helmet from 'react-helmet';

import Banner from './Banner';
import About from './About';
import Footer from './Footer';

import './_styles.scss';

const link = [{
  rel: 'stylesheet',
  href: 'https://fonts.googleapis.com/css?family=Montserrat:700',
}];

const Home = () => (
  <section className="home">
    <Helmet link={link} />
    <Banner />
    <About />
    <Footer />
  </section>
);

export default Home;
