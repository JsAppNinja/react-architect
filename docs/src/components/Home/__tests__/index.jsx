/* eslint-env jest */
import React from 'react';
import { createRouterSnapshot } from 'utils/testing';

import Home from '../';

describe('Home', () => {
  it('should render correctly', () => {
    const tree = createRouterSnapshot(<Home />);
    expect(tree).toMatchSnapshot();
  });
});
