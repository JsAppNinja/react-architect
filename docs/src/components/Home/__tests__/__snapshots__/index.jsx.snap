// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<section
  class="home"
>
  <header
    class="home__banner"
  >
    <h1
      class="md-display-2"
    >
      react-md
    </h1>
    <div
      class="home__logo"
    >
      some static content
    </div>
    <a
      class="md-btn md-btn--raised md-btn--text md-pointer--hover md-background--secondary md-background--secondary-hover md-inline-block"
      href="/components"
    >
      <div
        class="md-ink-container"
      >
        <span
          aria-hidden="true"
        />
      </div>
      View Demo
    </a>
  </header>
  <h3
    class="md-text-container home__about"
  >
    This project's goal is to be able to create a fully accessible material design styled website using React Components and Sass. With the separation of styles in Sass instead of inline styles, it should hopefully be easy to create custom components with the existing styles.
  </h3>
  <footer
    class="md-grid home__footer"
  >
    <a
      class="md-paper md-paper--1 md-card md-background--card md-cell home__svg-card"
      href="/getting-started"
    >
      <div
        class="md-card-title md-card-title--primary"
      >
        <h2
          class="md-card-title--title md-card-title--large md-text"
        >
          Getting Started
        </h2>
      </div>
      <section
        class="md-media md-media--1-1"
      >
        <img
          alt="A Mac and a Macbook with design mock ups."
          src="some static content"
        />
      </section>
    </a>
    <a
      class="md-paper md-paper--1 md-card md-background--card md-cell home__svg-card"
      href="/customization"
    >
      <div
        class="md-card-title md-card-title--primary"
      >
        <h2
          class="md-card-title--title md-card-title--large md-text"
        >
          Customization
        </h2>
      </div>
      <section
        class="md-media md-media--1-1"
      >
        <img
          alt="A chemistry set with material design colors flowing into a text editor."
          src="some static content"
        />
      </section>
    </a>
    <a
      class="md-paper md-paper--1 md-card md-background--card md-cell home__svg-card"
      href="/components"
    >
      <div
        class="md-card-title md-card-title--primary"
      >
        <h2
          class="md-card-title--title md-card-title--large md-text"
        >
          Components
        </h2>
      </div>
      <section
        class="md-media md-media--1-1"
      >
        <img
          alt="A chemistry set with material design colors flowing into a text editor."
          src="some static content"
        />
      </section>
    </a>
  </footer>
</section>
`;
